src/cli/arg.rs

```diff
@@ -0,0 +1,354 @@
+use std::fmt::Display;
```

Comment:
To me this very much looks like something which might exist in `clap`. Probably, at a lower level than the declarative API they advertise. Are we sure cannot use `clap` (or a similar crate) for this?